### DISCUSSION TITLE
Inconsistent socket emit while running multiple instances on docker

### DISCUSSION THREAD

**Discussion by codingram** [miguelgrinberg/Flask-SocketIO#1592] [Q&A]
We have a chatting application running on flask-socket-io. In this app, we have a end user, agent and manager.
Manager can see and reply all agents chats, agent can see and reply only the users connected to him/her. In the architecture, we are facing a issue where sometimes agent or manager does not receive any chats/messages emitted on room_id but when we refresh and rejoin manager/agent to same room_id, it works as expected. We checked the code and it looks fine. Most of the time, everything works as expected but we face this issue inconsistently. Is it the problem with architecture?

![github backend issue](https://user-images.githubusercontent.com/45767163/110781560-12f4d680-828c-11eb-8481-d46a2303106b.jpg)

![github issue frontend](https://user-images.githubusercontent.com/45767163/110781494-f6f13500-828b-11eb-90ff-824ca67a734f.jpg)

```
import json
import os
import requests

# importing eventlet for monkey_patching
import eventlet
eventlet.monkey_patch()

from datetime import datetime
from flask import Flask, jsonify, request, send_file
from flask_cors import CORS
from flask_socketio import SocketIO, emit, join_room, send
from flask_celery import make_celery
from dateutil import parser
from celery.task.control import revoke
from celery.task.control import broadcast
from celery.result import AsyncResult

app = Flask(__name__)
app.config["SECRET_KEY"] = Config.SECRET_KEY
app.config['CELERY_BROKER_URL'] = os.environ.get('rabbit_path')
app.config['CELERY_BACKEND'] = os.environ.get('rabbit_path')

CORS(app)
celery = make_celery(app)

socketio = SocketIO(app, cors_allowed_origins="*", message_queue=os.environ.get('REDIS_URL'))


if __name__ == "__main__":
    socketio.run(app, debug=debug_flag, host="0.0.0.0", port=port)
```

**Comment by miguelgrinberg** [comment]
You haven't provided the logs for a section of time where the problem exists.
